sdk/core/Azure.Core/src/Internal/BitVector640.cs

```diff
@@ -0,0 +1,165 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+
+namespace Azure.Core.Internal;
+
+/// <summary>
+/// This type effectively stores 640 bool values, but compresses their storage
+/// into ten ulongs, where each bit of the ulong represents a single bool value.
+///
+/// It exposes a public indexer so that the bool values can be accessed as a
+/// standard .NET collection API.
+///
+/// It is used in System.ClientModel and Azure.Core to implement response
+/// classifiers, where each true bit represents a status code that the
+/// classifier considers a success code.
+/// </summary>
+internal struct BitVector640
+{
+    // Keeping ulongs as fields puts them on the stack.
+    private ulong _bits0;
+    private ulong _bits1;
+    private ulong _bits2;
+    private ulong _bits3;
+    private ulong _bits4;
+    private ulong _bits5;
+    private ulong _bits6;
+    private ulong _bits7;
+    private ulong _bits8;
+    private ulong _bits9;
+
+    public bool this[int i]
+    {
+        readonly get
```

Comment:
That's interesting.   I don't know that I've ever seen this modifier pattern before.